Various missing salt module unit tests for 2.5+

TeddyAndrieux opened the pull request:
**Component**:

'salt', 'tests'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

#2266 

**Summary**:

- unit tests for `metalk8s.format_slots`
- Fix `metalk8s_kubernetes.get_object_digest` for dict
- unit tests for `metalk8s_kubernetes.get_object_digest`
- unit tests for `metalk8s_network.get_oidc_service_ip`

**Acceptance criteria**: 

```
---------- coverage: platform linux2, python 2.7.18-final-0 ----------
Name                                   Stmts   Miss  Cover   Missing
--------------------------------------------------------------------
salt/_modules/metalk8s.py                150      0   100%
salt/_modules/metalk8s_kubernetes.py     147      0   100%
salt/_modules/metalk8s_network.py         26      0   100%
--------------------------------------------------------------------
TOTAL                                    323      0   100%
```

---

Refs: #2266

TeddyAndrieux committed:
salt/tests: Add unit tests for `format_slots` salt function

Add unit tests for `format_slots` function from `metalk8s`
salt custom module

Refs: #2266

TeddyAndrieux committed:
salt/tests: Add unit tests for `get_object_digest` salt function

Add unit tests for `get_object_digest` function from
`metalk8s_kubernetes` salt custom module

Refs: #2266

TeddyAndrieux committed:
salt/tests: Add unit tests for `get_oidc_service_ip` salt function

Add unit tests for `get_oidc_service_ip` function from
`metalk8s_network` salt custom module

Refs: #2266

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.6/improvement/GH-2266-add-missing-salt-module-unit-tests-25` with contents from `improvement/GH-2266-add-missing-salt-module-unit-tests-25`
and `development/2.6`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.6/improvement/GH-2266-add-missing-salt-module-unit-tests-25 origin/development/2.6
 $ git merge origin/improvement/GH-2266-add-missing-salt-module-unit-tests-25
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/improvement/GH-2266-add-missing-salt-module-unit-tests-25
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

TeddyAndrieux committed:
salt: Dump dict sorted to compute digest in `metalk8s_kubernetes`

In `metalk8s_kubernetes.get_object_digest` we compute the digest of
object but in python2 when we dump dict as string with `str` the dict
order may change, so the digest may also change with it, to fix this,
just dump the dict using `json.dumps` with `sort_keys=True`

bert-e commented:
# History mismatch



Merge commit #2e90b26c661997c5ec675dd549333e421b508b92 on the integration branch
`w/2.6/improvement/GH-2266-add-missing-salt-module-unit-tests-25` is merging a branch which is neither the current
branch `improvement/GH-2266-add-missing-salt-module-unit-tests-25` nor the development branch
`development/2.6`.

It is likely due to a rebase of the branch `improvement/GH-2266-add-missing-salt-module-unit-tests-25` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

alexandre-allard reviewed (approved):
LGTM

TeddyAndrieux commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue GH-2266.

Goodbye teddyandrieux.
